include/mbedtls/md.h

```diff
@@ -337,6 +339,8 @@ MBEDTLS_CHECK_RETURN_TYPICAL
 int mbedtls_md(const mbedtls_md_info_t *md_info, const unsigned char *input, size_t ilen,
                unsigned char *output);
 
+#if defined(MBEDTLS_DECLARE_PRIVATE_IDENTIFIERS)
```

Comment:
Isn't the case that we should also in `md.c` replace the `#if defined(MBEDTLS_MD_C)` by `#if defined(MBEDTLS_DECLARE_PRIVATE_IDENTIFIERS)`? @mpg wdyt?

Reply:
No: `MBEDTLS_DECLARE_PRIVATE_IDENTIFIERS` guards things that are needed in the build, but not meant to be consumed by applications. These functions are still used in other library modules, possibly also in mbedtls as well, so they must be in the build.

Reply:
Specifically regarding `setup()`, when it's called by an application (that doesn't define `PRIVATE_IDENTIFIERS` before including `md.h`) in a build with `MD_C` defined, the application should always be calling it with the `hmac` argument being `0`, but this will no longer be enforced by the code.

I'm not sure that's a big problem though. If the application is really trying to use HMAC, it will get compiler errors when trying to call `mbedtls_md_hmac()` functions. If it's not trying to use HMAC, but it's still passing `hmac == 1`, then (1) the application is doing something stupid and (2) the consequences are only and extra allocation that will never be used. So, I think that's quite acceptable.

Reply:
I guess the task description wasn't very clear if you weren't immersed in the history of md-light. I've made it more explicit in https://github.com/Mbed-TLS/TF-PSA-Crypto/blob/a84c1e70547bcbe5e380235d3e4a86032dab8670/docs/architecture/0e-plans.md#public-hash-only-mdh